### ISSUE TITLE
`sbt console` repl sessions throwing `java.lang.ClassNotFoundException` when constructing `AnyVal`s since `3.2.1`

### ISSUE THREAD

**Issue by dzanot** [lampepfl/dotty#17083]
Repro steps
```
> sbt "new scala/scala3.g8 --name=repro"
...
> cd repro
> echo "package com.acme" > src/main/scala/Thing.scala
> echo "final case class Thing(i: Int) extends AnyVal" >> src/main/scala/Thing.scala
> sbt console
[info] welcome to sbt 1.8.2 (AdoptOpenJDK Java 11.0.9)
[info] loading settings for project global-plugins from plugins.sbt ...
[info] loading global plugins from /Users/dzanot/.sbt/1.0/plugins
[info] loading project definition from /Users/dzanot/repos/dzanot/repro/project
[info] loading settings for project root from build.sbt ...
[info] set current project to repro (in build file:/Users/dzanot/repos/dzanot/repro/)
[info] compiling 2 Scala sources to /Users/dzanot/repos/dzanot/repro/target/scala-3.2.2/classes ...
Welcome to Scala 3.2.2 (11.0.9, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                                                                                                                                                                                                                                              
scala> com.acme.Thing(1)
[error] java.lang.ClassNotFoundException: Thing
[error] 	at dotty.tools.repl.AbstractFileClassLoader.findClass(AbstractFileClassLoader.scala:51)
[error] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
[error] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
[error] 	at dotty.tools.repl.AbstractFileClassLoader.loadClass(AbstractFileClassLoader.scala:57)
[error] 	at java.base/java.lang.Class.forName0(Native Method)
[error] 	at java.base/java.lang.Class.forName(Class.java:398)
[error] 	at dotty.tools.repl.Rendering.rewrapValueClass(Rendering.scala:135)
[error] 	at dotty.tools.repl.Rendering.$anonfun$4(Rendering.scala:114)
[error] 	at scala.Option.flatMap(Option.scala:283)
[error] 	at dotty.tools.repl.Rendering.valueOf(Rendering.scala:114)
[error] 	at dotty.tools.repl.Rendering.renderVal(Rendering.scala:157)
[error] 	at dotty.tools.repl.ReplDriver.$anonfun$7(ReplDriver.scala:365)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.immutable.List.foreach(List.scala:333)
[error] 	at dotty.tools.repl.ReplDriver.extractAndFormatMembers$1(ReplDriver.scala:371)
[error] 	at dotty.tools.repl.ReplDriver.renderDefinitions$$anonfun$2(ReplDriver.scala:403)
[error] 	at scala.Option.map(Option.scala:242)
[error] 	at dotty.tools.repl.ReplDriver.renderDefinitions(ReplDriver.scala:409)
[error] 	at dotty.tools.repl.ReplDriver.compile$$anonfun$2(ReplDriver.scala:307)
[error] 	at scala.util.Either.fold(Either.scala:189)
[error] 	at dotty.tools.repl.ReplDriver.compile(ReplDriver.scala:323)
[error] 	at dotty.tools.repl.ReplDriver.interpret(ReplDriver.scala:248)
[error] 	at dotty.tools.repl.ReplDriver.loop$1(ReplDriver.scala:168)
[error] 	at dotty.tools.repl.ReplDriver.runUntilQuit$$anonfun$1(ReplDriver.scala:171)
[error] 	at dotty.tools.repl.ReplDriver.withRedirectedOutput(ReplDriver.scala:191)
[error] 	at dotty.tools.repl.ReplDriver.runBody$$anonfun$1(ReplDriver.scala:179)
[error] 	at dotty.tools.runner.ScalaClassLoader$.asContext(ScalaClassLoader.scala:80)
[error] 	at dotty.tools.repl.ReplDriver.runBody(ReplDriver.scala:179)
[error] 	at dotty.tools.repl.ReplDriver.runUntilQuit(ReplDriver.scala:171)
[error] 	at xsbt.ConsoleInterface.run(ConsoleInterface.java:52)
[error] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[error] 	at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error] 	at sbt.internal.inc.AnalyzingCompiler.console(AnalyzingCompiler.scala:233)
[error] 	at sbt.Console.console0$1(Console.scala:64)
[error] 	at sbt.Console.$anonfun$apply$5(Console.scala:74)
[error] 	at sbt.Run$.executeSuccess(Run.scala:186)
[error] 	at sbt.Console.$anonfun$apply$4(Console.scala:74)
[error] 	at sbt.internal.util.Terminal.withRawInput(Terminal.scala:145)
[error] 	at sbt.internal.util.Terminal.withRawInput$(Terminal.scala:143)
[error] 	at sbt.internal.util.Terminal$ProxyTerminal$.withRawInput(Terminal.scala:392)
[error] 	at sbt.Console.$anonfun$apply$3(Console.scala:74)
[error] 	at sbt.internal.util.Terminal$TerminalImpl.withRawOutput(Terminal.scala:990)
[error] 	at sbt.internal.util.Terminal$ProxyTerminal$.withRawOutput(Terminal.scala:431)
[error] 	at sbt.Console.apply(Console.scala:71)
[error] 	at sbt.Console.apply(Console.scala:49)
[error] 	at sbt.Console.apply(Console.scala:41)
[error] 	at sbt.Defaults$.$anonfun$consoleTask$1(Defaults.scala:2229)
[error] 	at sbt.Defaults$.$anonfun$consoleTask$1$adapted(Defaults.scala:2215)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:834)
[error] (Compile / console) java.lang.ClassNotFoundException: Thing
[error] Total time: 5 s, completed Mar 10, 2023, 3:08:20 PM
```
Tested and working fine in with `3.2.0`

If the anyval is defined within the repl session, or without a `package` it works fine.

**Comment by SethTisue** [comment]
Also reproducible with scala-cli:

```
% cat Thing.scala
//> using scala "3.2.2"
package com.acme
final case class Thing(i: Int) extends AnyVal
% scala-cli repl .
Starting compilation server
Compiling project (Scala 3.2.2, JVM)
Compiled project (Scala 3.2.2, JVM)
Welcome to Scala 3.2.2 (17.0.6, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                    
scala> com.acme.Thing(1)
Exception in thread "main" java.lang.ClassNotFoundException: Thing
	at dotty.tools.repl.AbstractFileClassLoader.findClass(AbstractFileClassLoader.scala:51)
...
```

**Comment by SethTisue** [comment]
and with command-line `scalac`, with `/usr/local/scala/scala3-3.2.2/bin/scalac Thing.scala` followed by `/usr/local/scala/scala3-3.2.2/bin/scala -classpath .`

**Comment by SethTisue** [comment]
oh, it's fixed as of 3.3.0-RC3
